src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> | object {
+  if (items.length === 0) {
+    return {};
+  }
+
+  const result: GroupsMap<T> = {};
+
+  for (const item of items) {
+    const groupKey = item[key] as string;
```

Comment:
There is an assumption here that the `key` property of `item` will always be of type `string`. However, if `key` is not a string, this could lead to unexpected behavior. It would be safer to check the type of `item[key]` and handle cases where it is not a string.